back-end/www/models/model_operations/location_operations.py

```diff
@@ -0,0 +1,278 @@
+    Set the current time to done_at to mark it is done.
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+    is_done : bool
+        Set done or not done.
+
+    Returns
+    -------
+    location : Location
+        The retrieved location object.
+
+    Raises
+    ------
+    exception : Exception
+        When is_done is not bool
+    exception : Exception
+        When no location is found.
+    """
+    if not isinstance(is_done, bool):
+        raise Exception("is_done shall be bool")
+
+    location = Location.query.filter_by(id=location_id).first()
+
+    if location is None:
+        raise Exception("No location found in the database to update.")
+
+    if is_done:
+       location.done_at = datetime.datetime.now()
+    else:
+       location.done_at = None
+
+    db.session.commit()
+    return location
+
+
+def remove_location(location_id):
+    """
+    Remove a location.
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+
+    Raises
+    ------
+    exception : Exception
+        When no location is found.
+    """
+    location = Location.query.filter_by(id=location_id).first()
+
+    if location is None:
+        raise Exception("No location found in the database to delete.")
+
+    db.session.delete(location)
+    db.session.commit()
+
+
+def get_locations(user_client_id, size, gold_standard_size):
+    """
+    Get the locations that can be returned to the front-end.
+
+    Parameters
+    ----------
+    user_client_id : str
+        client id of the user
+    size : int
+        Total number of locations to be returned.
+    gold_standard_size : int
+        The number of locations that should include gold standard answers.
+        There should be ("size" - "gold_standard_size") locations that are not labeled yet.
+
+    Returns
+    -------
+    locations : list of Locations
+        The list of retrieved location objects.
+
+    Raises
+    ------
+    exception : Exception
+        When size and gold_standard_size are not integers (or < 0).
+    exception : Exception
+        When gold standard does not exist.
+    exception : Exception
+        When gold standard size exceeds size.
+    exception : Exception
+        When we cannot find "gold_standard_size" of locations which have gold standards.
+    exception : Exception
+        When we cannot find "size" of locations.
+    """
+    if not isinstance(gold_standard_size, int):
+        raise Exception("The gold_standard_size shall be an integer")
+    if not isinstance(size, int):
+        raise Exception("The gold_standard_size shall be an integer")
+    if size < 0:
+        raise Exception("The size must be greater or equal to 0.")
+    if gold_standard_size < 0:
+        raise Exception("The gold_standard_size must be greater or equal to 0.")
+    if gold_standard_size > size:
+        raise Exception("The gold standard size cannot exceed size.")
+
+    if size == 0:
+        return None
```

Comment:
Should return an empty list [] instead of None, right? Just to be consistent with the final retuned list.